### DISCUSSION TITLE
Alternative Screens and supported code

### DISCUSSION THREAD

**Discussion by ArcdoesMC** [Ralim/IronOS#834] [Ideas]
I thought it would be a good idea to look for different screen types to replace the stock oled. It looks like miniware puts the cheapest possible screens in their products. I got the ts80 and a es121 less than 2 years ago and after a few months of INUSE my ts80 is almost too dim to read and my es121 has the same issue but with the addition of dead, dim, and normal pixels. 

I don't think e-ink wouldn't work due to the driving requirements (it needs to produce high voltage) and low refresh rate, but let me know if you think otherwise. So I think TFT LCDs would be the best bet. But maybe someone has other ideas. I don't like that it looks like all these screens have a >2 year lifespan.

**Comment by Ralim** [comment]
I've found just replacing the OLED screen with a new one of the same type but in the white models to turn out ok and last a while.
I have one TS100 from 2016 on its original screen that is totally fine, but one from the same batch a friend had failed early. Suspect it may be to do with storage temperature humidity as to how long they last.

Main restriction is that you need the same pinout and I2C which really limits you to OLED.

It would be technically possible to put a small ips screen in there with an adapter board to translate I2C <-> Screen but its a lot of work.
